packages/cli/src/plugin.js

```diff
@@ -0,0 +1,256 @@
+import fs from 'fs'
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { getConfig, getPaths } from './lib'
+import { installModule } from './lib/packages'
+
+const PLUGIN_CACHE_FILENAME = 'command-cache.json'
+
+function checkPluginListAndWarn(plugins) {
+  // Plugins must define a package and version
+  for (const plugin of plugins) {
+    if (!plugin.package) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  A plugin is missing a package name, it cannot be loaded.`
+        )
+      )
+    }
+    if (!plugin.version) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  Plugin "${plugin.package}" is missing a version, it cannot be loaded.`
+        )
+      )
+    }
+  }
+
+  // Plugins should only occur once in the list
+  const pluginNames = plugins.map((p) => p.package)
+  if (plugins.length !== new Set(pluginNames).size) {
+    console.warn(
+      chalk.yellow(
+        '⚠️  Duplicate plugin names found in redwood.toml, duplicates will be ignored.'
+      )
+    )
+  }
+
+  // Plugins should be published to npm under a scope which is used as the namespace
+  const namespaces = plugins.map((p) => p.package.split('/')[0])
```

Comment:
There's a chance you can't call `split` on package here since you checked if `package` exists earlier and if it doesn't, it warns. (This is where TS would be invaluable of course. Maybe some JS doc to get by for now, but again it's early)

```suggestion
  const namespaces = plugins.map((p) => p.package.split('/')[0])
```